src/components/shared/Entity/ExistingEntityCards/Store/Hydrator.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+    useExistingEntity_hydrated,
+    useExistingEntity_hydrateState,
+    useExistingEntity_setHydrated,
+    useExistingEntity_setHydrationErrorsExist,
+} from 'components/shared/Entity/ExistingEntityCards/Store/hooks';
+import { useEntityType } from 'context/EntityContext';
+import useGlobalSearchParams, {
+    GlobalSearchParams,
+} from 'hooks/searchParams/useGlobalSearchParams';
+import { useEffect } from 'react';
+import { BaseComponentProps } from 'types';
+
+function ExistingEntityHydrator({ children }: BaseComponentProps) {
+    const connectorId = useGlobalSearchParams(GlobalSearchParams.CONNECTOR_ID);
+    const entityType = useEntityType();
+
+    const hydrated = useExistingEntity_hydrated();
+    const setHydrated = useExistingEntity_setHydrated();
+
+    const setHydrationErrorsExist = useExistingEntity_setHydrationErrorsExist();
+
+    const hydrateState = useExistingEntity_hydrateState();
+
+    useEffect(() => {
+        if (
+            !hydrated &&
+            connectorId &&
+            (entityType === 'capture' || entityType === 'materialization')
+        ) {
+            hydrateState(entityType, connectorId).then(
+                () => {
+                    setHydrated(true);
+                },
+                () => {
+                    setHydrated(true);
+                    setHydrationErrorsExist(true);
+                }
+            );
+        }
+    }, [
+        hydrateState,
+        setHydrated,
+        setHydrationErrorsExist,
+        connectorId,
+        entityType,
+        hydrated,
+    ]);
+
+    return <div>{children}</div>;
```

Comment:
We don't need the `div` here. We can use the empty tags and throw an eslint comment

Reply:
Will do. Every developer has a unique threshold for explicit, linting overrides that are deemed _acceptable_, so I tend to err on the side of caution when working in collaborative codebases.